### ISSUE TITLE
Anatomy of an Array/Memory Layout <-- maybe make it clearer?

### ISSUE THREAD

**Issue by MikeAnblips** [rougier/from-python-to-numpy#96]
Hi,
sorry I'm not 100% sure this is the right place to suggest that, but the following code can result a little unclear because you reference a variable `index` which is not previously defined:

![image](https://user-images.githubusercontent.com/20505054/93716018-38bae180-fb6d-11ea-8e83-ca1fa428057b.png)
Then, when double-checking whether the idea works, I think you mix up variable names a little:

![image](https://user-images.githubusercontent.com/20505054/93716057-7cade680-fb6d-11ea-8017-5711fe9e150d.png)

Not a big deal but I think a 'fusion' of the two snippets into one would be clearer, as direct implementation of what you propose does not work.

Best,
M

**Comment by rougier** [comment]
Yes, this is the perfect place to report the issue. Thanks a lot and I think you're right that there's a problem. Could you make a PR to fix it?

**Comment by MikeAnblips** [comment]
I will... As soon as I learn how to do it properly :)
Sorry for being such a noob, I'll look into it when I get a quiet 1/2h today.

**Comment by rougier** [comment]
No problem. Just ask me if you need some help (if this is your first pull request, it might appear a bit complex)

**Comment by MikeAnblips** [comment]
Thanks Nicolas,
indeed I need some help. When trying to issue a new PR, I should input the reference and the modified branch, which I understand should be your base and my 'proposal', respectively. Is this understanding correct?
If yes, I don't know how to create the proposal for change. I'm trying to work out what has been done for another PR I can see:
![image](https://user-images.githubusercontent.com/20505054/94234588-3dcfb600-ff0a-11ea-927f-688655ddd829.png)
So there's a txt file which has been modified in a diagram about the location of a pointer. The txt file is a chapter of your book I surmise... Shall I do the same action? Propose a mod version of your chapter?
If yes, how do I do that?

**Comment by rougier** [comment]
The easiest way, when the change is small is to browse the file on my repository and to "edit" it (the pen button on upper right). Since you do not have rights to modify the file, GitHub will propose you automatically to make a PR for proposing your change.

The other way is more complex but not that much:
1. Fork the repository in you GtiHub account
2. Clone your new forked directory on you computer
3. cd to to you directoy and create new branch (`git checkout -b "corrections"` for example)
4. Do some change to the files, add them for the next commit (`git add`)
5. Do the actual commit (`git commit -m "message"`)
6. Push your changes (`git push`)

And you're mostly done.  If you go on GitHub gain, you'll see your last commit and GitHub will propose you to make a PR.

If it's not clear, just tell me. I remember that I totally messed up my first PR a few years ago and someone helped me a lot)
